### PR TITLE
[AERIE-2066] Report activity argument validations in `activity_directive_validations`

### PR DESCRIPTION
**Pull request by pcrosemurgy**
* **Tickets addressed:** AERIE-2066
* **Review:** By file  
* **Merge strategy:** Merge (no squash)  

## Description
Adds a Hasura event trigger triggered off of inserts/updates to `activity_directive.arguments` to update the `activity_directive_validations` table.

Suggested to review by file since some refactoring commits were layered on top of each other to incrementally verify PGSQL & Hasura metadata correctness.

## Verification

Upon creation of a `BakeBananaBread` activity with valid arguments the `activity_directive` table now contains:
Screen Shot 2022-09-16 at 6 47 30 PM

Upon creation of a `BakeBananaBread` activity with invalid arguments (a `temperature` of `-2`) the associated `activity_directive_validations.validations` now contains:
```json
{
    "errors": [
        {
            "message": "Temperature must be positive",
            "subjects": [
                "temperature"
            ]
        }
    ],
    "success": false
}
```

## Documentation
TODO will start documentation when the approach made in this PR is roughly accepted.

## Future work
- UI will make use of this new column (@camargo).
- Should we consider removing the `validateActivityArguments` Hasura action since the UI will no longer rely on this action?
- Maybe we should consider adding a `mission_model_parameters.validations` column for symmetry.
  - This could also remove the need for a `validateModelArguments` Hasura action.

### CODE REVIEW

**Review by camargo — APPROVED** [review]
This is great. Tested locally and observed the validations. Also I'm all for adding the `mission_model_parameters.validations` for API consistency. Once we have both I think we can safely remove the actions before 1.0.

**Comment by pcrosemurgy** [comment]
Thanks for testing @camargo. I'll go ahead a make a ticket for adding a `mission_model_parameters.validations` (and event trigger) to replace the Hasura action.

File: merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/http/MerlinBindings.java [hunk 1]
```diff
@@ -19,12 +19,7 @@
 import java.util.Map;
 import java.util.stream.Collectors;
 
-import static gov.nasa.jpl.aerie.merlin.server.http.HasuraParsers.hasuraActivityActionP;
-import static gov.nasa.jpl.aerie.merlin.server.http.HasuraParsers.hasuraExternalDatasetActionP;
-import static gov.nasa.jpl.aerie.merlin.server.http.HasuraParsers.hasuraMissionModelActionP;
-import static gov.nasa.jpl.aerie.merlin.server.http.HasuraParsers.hasuraMissionModelArgumentsActionP;
-import static gov.nasa.jpl.aerie.merlin.server.http.HasuraParsers.hasuraMissionModelEventTriggerP;
-import static gov.nasa.jpl.aerie.merlin.server.http.HasuraParsers.hasuraPlanActionP;
+import static gov.nasa.jpl.aerie.merlin.server.http.HasuraParsers.*;
```
Comment (goetzrrGit) [inline merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/http/MerlinBindings.java]: I remember Jonathan wasn't a big fan of importing all packages with `.*` but only the ones we needed for the class

File: merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/remotes/postgres/PostgresParsers.java [hunk 2]
```diff
@@ -3,31 +3,76 @@
...
+import java.time.temporal.ChronoField;
 import java.util.Map;
 
-import static gov.nasa.jpl.aerie.json.BasicParsers.chooseP;
-import static gov.nasa.jpl.aerie.json.BasicParsers.intP;
-import static gov.nasa.jpl.aerie.json.BasicParsers.literalP;
-import static gov.nasa.jpl.aerie.json.BasicParsers.longP;
-import static gov.nasa.jpl.aerie.json.BasicParsers.mapP;
-import static gov.nasa.jpl.aerie.json.BasicParsers.productP;
+import static gov.nasa.jpl.aerie.json.BasicParsers.*;
```
Comment (goetzrrGit) [inline merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/remotes/postgres/PostgresParsers.java]: Same with this import

**Review by goetzrrGit — APPROVED** [review]
Looks good to me. Just a minor thing that I remember Jonathan mentioned in my PR once.

**Comment by pcrosemurgy** [comment]
Thanks Ryan! Just pushed static import changes in 63f80299aa565078f1e429fbac61a32febd6beb2 :)

File: merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/http/MerlinBindings.java (hunk 1 above)
Comment (patkenneally) [reply merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/http/MerlinBindings.java]: The project in general prefers explicit imports ;)

**Pull request merged**